tests/test_room.py

```diff
@@ -0,0 +1,17 @@
+from PyQuotient import Quotient
+from __feature__ import snake_case, true_property
+
+
+def test_init():
+    connection = Quotient.Connection()
+    room = Quotient.Room(connection, 'room1', Quotient.JoinState.Join)
+    assert isinstance(room, Quotient.Room)
+
+
+def test_subclass():
+    class PyRoom(Quotient.Room):
+        ...
+
+    connection = Quotient.Connection()
+    room = PyRoom(connection, 'room1', Quotient.JoinState.Join)
```

Comment:
This doesn't really show that you can get a room of a correct class from inside the `Connection` class, for now. The setup I'd expect here (but, as discussed, this is "nice to have" but not required for GSoC) is:
```suggestion
    connection.set_room_type(PyRoom)
    room = connection.provide_room('!room1:example.com', Quotient.JoinState.Join)
```

Reply:
Yes, I understand, but the goal of this test was to test whether creating an instance of a subclass of binding works.
As we discussed earlier it's not so easy to use the subclass in connection as you suggest, deeper investigation to find the right approach is needed. I can also create an issue to not forgot to do this later.